engines/ospray/ispc/geometry/utils/SafeIndex.ih

```diff
@@ -24,27 +24,36 @@
 
-#define __define_safe_index_internal(T, UNIFORM_VARYING, CONST)       \
-    inline CONST uniform T* UNIFORM_VARYING                           \
-        safe_index(const UNIFORM_VARYING bool useSafeIndex,           \
-                   CONST T* UNIFORM_VARYING const base,               \
-                   const UNIFORM_VARYING uint64 index)                \
-    {                                                                 \
-        if (useSafeIndex)                                             \
-        {                                                             \
-            const uniform uint64 tSize = sizeof(uniform T);           \
-            const UNIFORM_VARYING uint64 offsetBytes = tSize * index; \
-            return (CONST T * UNIFORM_VARYING const)((uint64)base +   \
-                                                     offsetBytes);    \
-        }                                                             \
-        else                                                          \
-        {                                                             \
-            return base + index;                                      \
-        }                                                             \
+#define INT_MAX 2147483647
+
+#define __define_safe_index_internal(T, UNIFORM_VARYING, CONST)               \
+    inline CONST uniform T* UNIFORM_VARYING safe_index(                       \
+        const uniform bool useSafeIndex, CONST T* UNIFORM_VARYING const base, \
+        const UNIFORM_VARYING uint64 index)                                   \
+    {                                                                         \
+        if (useSafeIndex)                                                     \
+        {                                                                     \
+            const uniform uint64 tSize = sizeof(uniform T);                   \
+            const UNIFORM_VARYING uint64 offsetBytes = tSize * index;         \
+            return (CONST T * UNIFORM_VARYING const)((uint64)base +           \
+                                                     offsetBytes);            \
+        }                                                                     \
+        else                                                                  \
+        {                                                                     \
+            return base + index;                                              \
+        }                                                                     \
+    }
+
+#define __define_needs_safe_index(T)                                         \
+    inline uniform bool needs_safe_index(const T* uniform ptr,               \
+                                         const uniform uint64 numGeometries) \
+    {                                                                        \
+        return (sizeof(uniform T) * numGeometries) >= INT_MAX;               \
```

Comment:
Following the code conventions, these two functions should be needsSafeIndex and safeIndex (which I would rename to safeDereference, but you may keep safeIndex). And the macro should be DEFINE_SAFE_INDEX (the double score makes it look like a compiler internal)

Reply:
Maybe it should be called safeIncrement since that is actually what it is doing, not dereferencing nor indexing.